CMakeLists.txt

```diff
@@ -117,8 +134,7 @@ add_library(picohttp-core
 )
 
 target_link_libraries(picohttp-core
-    ${PTLS_LIBRARIES}
-    ${OPENSSL_LIBRARIES}
+    picoquic-crypto-picotls-openssl
 )
```

Comment:
I don't like the idea of separate libraries for each crypto variant. That is making the project more complex, and requires a cascade of changes in applications.

Reply:
I thought of statically linking the crypto implementation. This should help seperating the core implementation from the crypto backend implementation and prevent any acidental usage of direct crypto calls. But i would have kept all crypto implementation in the same library for now. What do you think?